### ISSUE TITLE
[Feat] Improve Preserved Important Message in Notion  With Notion's Link Preview 

### ISSUE THREAD

**Issue by Nabhag8848** [RocketChat/Apps.Notion#39]
### What do we need?

- Currently when Preserving Message in Page or Subpage we are using Notion Block of type callout. 
- Api Docs: https://developers.notion.com/reference/block#callout
- https://github.com/RocketChat/Apps.Notion/blob/main/src/lib/NotionSDK.ts#L1005
- To Know What is Feature about: https://github.com/Nabhag8848/Google-Summer-Of-Code/?tab=readme-ov-file#10-preserve-message-in-notion-page
- We need to enhance using Link Preview of Notion. 
 https://developers.notion.com/reference/unfurl-attribute-object
 

### Acceptance Criteria

- [ ] Research how you can get this working.
- [ ] Research if is it possible in Self hosted RocketChat as every preview link has specific domain.

### Relevant ScreenShots

https://github.com/RocketChat/Apps.Notion/assets/65061890/acabbacf-d5f7-47d6-a219-a0061f205d08

**Comment by Spiral-Memory** [comment]
I am going through the documentation to understand how it works, and I would like to work on it.

**Comment by Nabhag8848** [comment]
1. not an issue, but noted, Do you think We Can write script  for reducing those steps for non rc cloud user.
2. for non rc cloud server this can be little blockage and for rc cloud as it will be rocket. chat they don't  need to fill out this form in that case for them as long as they don't have access we can show current implementation we have.
3. challenging +  long process.

Question: from step 2 and step 3 what i can see is if we need to make work in `open.rocket.chat` and `xyz.rocket.chat` we need to provide separetly tokens and fill form for both of them. if thats a case they this will cause every rocketchat workspace user using notion app to fill this form ?

**Comment by Spiral-Memory** [comment]
1. Writing scripts might not be possible, as only the workspace owner can access the integrations, and there is no API currently to access those settings from outside. Filling out the forms is in the integration section. However, good documentation could be provided to guide them on how to do it seamlessly.

Regarding your question: I feel that if both are using the same domain, then there won't be an issue. For the first time, the user just has to authorize it, and it can be globally set up, similar to how Slack does it for their workspaces, as there is no option to self-host it, and all workspaces run on a single domain.

However, for self-hosted instances, this has to be done for every workspace owner, and it has to be done for every domain.

**Comment by Spiral-Memory** [comment]
@Nabhag8848 , I haven't received API access for link preview yet. I've emailed the Notion team again. I'll let you know once I get access and test it.

**Comment by Nabhag8848** [comment]
@Spiral-Memory Any updates from Notion Team ?

**Comment by Spiral-Memory** [comment]
> @Spiral-Memory Any updates from Notion Team ?

Yes i got a reply, but it wasn't very helpful. Let me share you the screenshot.


![Screenshot_20240229-200907.png](https://github.com/RocketChat/Apps.Notion/assets/78961432/dfd4fe7f-c3a9-41a0-8a43-bdae57fed1b2)

**Comment by Spiral-Memory** [comment]
Yep 🥲

I have replied to them again that I know these details and please reply with the exact thing of what I'm asking.. let's see if they do.

**Comment by Spiral-Memory** [comment]
Hey @Nabhag8848 I got another reply today, have a look at it and provide me enough info to respond back to this email : 
![image](https://github.com/RocketChat/Apps.Notion/assets/78961432/4c5664d2-efb2-40fa-aa7c-fb955e8ce1e8)

**Comment by Nabhag8848** [comment]
> Hey @Nabhag8848 I got another reply today, have a look at it and provide me enough info to respond back to this email :
> 
> ![image](https://private-user-images.githubusercontent.com/78961432/311332039-4c5664d2-efb2-40fa-aa7c-fb955e8ce1e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NjUxMTgsIm5iZiI6MTcwOTk2NDgxOCwicGF0aCI6Ii83ODk2MTQzMi8zMTEzMzIwMzktNGM1NjY0ZDItZWZiMi00MGZhLWFhN2MtZmI5NTVlOGNlMWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDA2MTMzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzOTZmNjQ2NjY4Y2YzNGQ0NTI2OTUyZTc4ZWI1NDkyYjAwY2U1OTk4NTA2ODQwYzMyYzhhZDZhZWI1NjU4NWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vWxJ7s_iloJcbdp5vLwWL2MJk8HV2Ja66JvfBogjCxo)

This is awesome @Spiral-Memory , So Exicted for this one. Will share all the details by end of day. I will need to look at details. 

 - some thoughts rocketchat is supported for self-hosted and also cloud version rocketchat provides for users that instance goes to (https://<subdomain> .rocket.chat). We need to look if they will support self-hosted cause  their domain will be different.

**Comment by Spiral-Memory** [comment]
> > Hey @Nabhag8848 I got another reply today, have a look at it and provide me enough info to respond back to this email :
> > 
> > ![image](https://private-user-images.githubusercontent.com/78961432/311332039-4c5664d2-efb2-40fa-aa7c-fb955e8ce1e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NjUxMTgsIm5iZiI6MTcwOTk2NDgxOCwicGF0aCI6Ii83ODk2MTQzMi8zMTEzMzIwMzktNGM1NjY0ZDItZWZiMi00MGZhLWFhN2MtZmI5NTVlOGNlMWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDA2MTMzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzOTZmNjQ2NjY4Y2YzNGQ0NTI2OTUyZTc4ZWI1NDkyYjAwY2U1OTk4NTA2ODQwYzMyYzhhZDZhZWI1NjU4NWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vWxJ7s_iloJcbdp5vLwWL2MJk8HV2Ja66JvfBogjCxo)
> 
> This is awesome @Spiral-Memory , So Exicted for this one. Will share all the details by end of day. I will need to look at details. 
> 
>  - some thoughts rocketchat is supported for self-hosted and also cloud version rocketchat provides for users that instance goes to (https://<subdomain> .rocket.chat). We need to look if they will support self-hosted cause  their domain will be different. 

Sure @Nabhag8848 Take your time 🤗
I'm also very excited for it !

**Comment by Spiral-Memory** [comment]
Hey @Nabhag8848 I got an automated reply that they are waiting for the response. Could you please provide all details as early as possible so that i can forward it to them. 
Thanks !

**Comment by Spiral-Memory** [comment]
Hey @Nabhag8848 I got an automated reply that they are waiting for the response. Could you please provide all details as early as possible so that i can forward it to them. 
Thanks !
